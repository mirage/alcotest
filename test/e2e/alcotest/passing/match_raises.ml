let to_test () = raise (Failure "")
let expect_failure = function Failure _ -> true | _ -> false
let test () = Alcotest.match_raises "Generates Failure" expect_failure to_test

let () =
  Alcotest.run "Exceptions"
    [
      ( "matches_raises",
        [
          Alcotest.test_case "True means the exception is expected" `Quick test;
        ] );
    ]
