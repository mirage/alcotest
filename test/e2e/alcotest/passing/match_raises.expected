Testing `Exceptions'.
This run has ID `<uuid>'.

  [OK]          matches_raises          0   True means the exception is expec...

Full test results in `<build-context>/_build/_tests/<test-dir>'.
Test Successful in <test-duration>s. 1 test run.
